### PR TITLE
add a debugTrace to tracer configs

### LINKED ISSUE #144: make debug-logging tracer configurable
Finagle's `com.twitter.finagle.tracing.debugTrace=true` flag is very useful for debugging linkerd behavior locally.  Support this in the `tracers` configuration.

### PR DESCRIPTION
**Pull request by siggy**
this overrides finagle's com.twitter.finagle.tracing.debugTrace flag
fixes #144

### CODE REVIEW

File: linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala [hunk 1]
```diff
@@ -78,7 +78,11 @@ object Linker {
       // At least one router must be specified
       if (routers.isEmpty) throw NoRoutersSpecified
 
-      val tracer: Tracer = tracers.map(_.map(_.newTracer())) match {
+      val tracer: Tracer = tracers.map(_.map(t => {
+        // override the global {com.twitter.finagle.tracing.debugTrace} flag
+        fDebugTrace.parse(t.debugTrace.toString)
+        t.newTracer()
+      })) match {
```
Comment (olix0r) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala]: prefer `(_.map { t => ... })`

File: linkerd/docs/config.md [hunk 2]
```diff
@@ -414,6 +414,8 @@ configuring tracers via a top-level `tracers` section:
 
 * *tracers* --
   * *kind* -- One of the supported tracers, by fully-qualified class name.
+  * *debugTrace* -- Print all traces to the console. Note this overrides the global
+    `-com.twitter.finagle.tracing.debugTrace` flag. (default: false)
```
Comment (adleong) [inline linkerd/docs/config.md]: default is actually to fall back to the debugTrace flag, isn't it?

File: linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala [hunk 3]
```diff
@@ -78,7 +78,11 @@ object Linker {
       // At least one router must be specified
       if (routers.isEmpty) throw NoRoutersSpecified
 
-      val tracer: Tracer = tracers.map(_.map(_.newTracer())) match {
+      val tracer: Tracer = tracers.map(_.map(t => {
```
Comment (adleong) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala]: style nit:
```
val tracer: Tracer = tracers.map(_.map { t =>
```

File: linkerd/docs/config.md (hunk 2 above)
Comment (siggy) [inline linkerd/docs/config.md]: yeah, where debugTrace itself defaults to false. wasn't sure how explicit to be here. i'll update.

File: linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala (hunk 3 above)
Comment (siggy) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala]: lol u guys https://github.com/BuoyantIO/linkerd/pull/354#discussion_r61657709

**Comment by rmars** [comment]
:star: woww nice.  idk if you want to call out the new config file in changes.md

**Comment by adleong** [comment]
👍 looks good! could you also update config.md?

**Comment by adleong** [comment]
:star: 🐞

**Pull request merged**